libraries/WebServer/src/WebServer.cpp

```diff
@@ -796,3 +803,13 @@ String WebServer::_responseCodeToString(int code) {
     default:  return F("");
   }
 }
+
+#if SOC_WIFI_SUPPORTED
+  bool ON_STA_FILTER(WebServer &server) {
+    return WiFi.STA.localIP() == server.client().localIP();
```

Comment:
```cpp
return WiFi.STA.hasIP() && WiFi.STA.localIP() == server.client().localIP();
```

Else it could return IP 0 which will also match with the client IP being 0

Reply:
Done